src/main/java/com/github/adamyork/wiremock/transformer/VelocityResponseTransformer.java

```diff
@@ -35,40 +45,67 @@
-                      context = toolManager.createContext();
-                      addBodyToContext(request.getBodyAsString());
-                      addHeadersToContext(request.getHeaders());
-                      context.put("requestAbsoluteUrl", request.getAbsoluteUrl());
-                      context.put("requestUrl", request.getUrl());
-                      context.put("requestMethod", request.getMethod());
-                      final String body = getRenderedBody(responseDefinition);
-                      return ResponseDefinitionBuilder.like(responseDefinition).but()
-                              .withBody(body)
-                              .build();
-                } else {
-                    return responseDefinition;
+                                        final ResponseDefinition responseDefinition, final FileSource files,
+                                        final Parameters parameters) {
+        if(responseDefinition.specifiesBodyFile() && templateDeclared(responseDefinition)) {
+            this.fileSource = files;
+            final VelocityEngine velocityEngine = new VelocityEngine();
+            velocityEngine.init();
+            final ToolManager toolManager = new ToolManager();
+            toolManager.setVelocityEngine(velocityEngine);
+            context = toolManager.createContext();
+            addBodyToContext(request.getBodyAsString());
+            addHeadersToContext(request.getHeaders());
+            addParametersVariablesToContext(request.getUrl());
+            context.put("requestAbsoluteUrl", request.getAbsoluteUrl());
+            context.put("requestUrl", request.getUrl());
+            context.put("requestMethod", request.getMethod());
+            final String body = getRenderedBody(responseDefinition);
+            return ResponseDefinitionBuilder.like(responseDefinition).but()
+                    .withBody(body)
+                    .build();
+        } else {
+            return responseDefinition;
+        }
+    }
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    /**
+     * @param url the request url to be parsed
+     */
+    private void addParametersVariablesToContext(final String url) {
+        try {
+            final Map<String, Integer> uniqueMap = new HashMap<>();
+            final List<NameValuePair> result = URLEncodedUtils.parse(new URI(url), Charset.defaultCharset());
+            for (NameValuePair nvp : result) {
+                int paramCount = 1;
+                if(uniqueMap.get(nvp.getName()) != null) {
+                    paramCount = uniqueMap.get(nvp.getName());
                 }
+                final String actualKey = nvp.getName().concat(Integer.toString(paramCount));
+                context.put(actualKey.replace("-", ""), nvp.getValue());
+                uniqueMap.put(nvp.getName(), ++paramCount);
+            }
+        } catch (final URISyntaxException e) {
+            e.printStackTrace();
+        }
```

Comment:
It looks good since I don't need to worry about extra parameter to pass but I'm little bit concerned to handle multiple values for one parameter name.  Basically, you add postfix as index like `xxx1, xx2, ...` and that doesn't allow iteration over that specific param.   Typical use case of using multiple valued param is with indefinite number of values.  I know put `[0]` at the end of usual parameter might bother readability and there could be better solution.

Reply:
Honestly your use case was a bit unusual. Normally you don't encounter an api that supports multiple declarations of the same parameter, i.e. product-code. Usually you would express this like , ;product-codes=1,2,3,4. Matrix parameters work better here. Default at 1th index of keys was in your example, but i'm not strongly opinionated one way or the other. Could just as easily start it at 0.

Reply:
Well, I do have such a service and try to mock it with wiremock.  It turns out wiremock naturally support it since the return type of Request::queryParameter is MultiValue which has one key and a list of values.  That's how I implemented it easily w/o any additional parsing or processing.  
Yes, if I designed the API, I would go with comma separated list but I already experienced several RESTful services doing this way and I had no choice. :(